FEXCore/Source/Interface/Core/JIT/Arm64/MemoryOps.cpp

```diff
@@ -578,6 +578,37 @@ ARMEmitter::ExtendedMemOperand Arm64JITCore::GenerateMemOperand(
   FEX_UNREACHABLE;
 }
 
+void Arm64JITCore::CalculateMemOperand(ARMEmitter::Register Destination, uint8_t AccessSize, ARMEmitter::Register Base,
+                                       IR::OrderedNodeWrapper Offset, IR::MemOffsetType OffsetType, uint8_t OffsetScale) {
+  if (Offset.IsInvalid()) {
```

Comment:
Remove this whole function and just assert(Offset.IsInvalid()) in the ASIMD path. The OpcodeDispatcher address selection won't generate nontrivial addressing if we don't want it to, it already has the logic to expand complex addressing into 64-bit ALU.

Reply:
Added the assert and remove this function. It actually shaved off two `mov` instructions from the implementation because of that.